connectors/uplynk/src/main/java/com/theoplayer/android/connector/uplynk/internal/UplynkAdIntegration.kt

```diff
@@ -137,11 +296,26 @@ internal class UplynkAdIntegration(
                 try {
                     val response = it.parseExternalResponse()
                     eventDispatcher.dispatchPreplayEvents(response)
-                    adScheduler = UplynkAdScheduler(response.ads.breaks, AdHandler(controller))
+                    adScheduler = UplynkAdScheduler(response.ads.breaks, AdHandler(controller, uplynkConfiguration.defaultSkipOffset))
                 } catch (e: Exception) {
                     eventDispatcher.dispatchPreplayFailure(e)
                     controller.error(e)
                 }
             }
     }
+
+    override fun skipAd(ad: Ad) {
+        if (isAdSkippable(ad)) {
+            adScheduler?.getSkipToTime(ad, player.currentTimeDurationSeconds())?.let {
+                seek(it)
+                adScheduler?.skipAd(ad)
+            }
+        }
+    }
+
+    private fun isAdSkippable(ad: Ad): Boolean = ad.skipOffset != -1
 }
+
+private fun timeDurationSeconds(time: Double) = time.toDuration(DurationUnit.SECONDS)
```

Comment:
This is equivalent to
```kotlin
import kotlin.time.Duration.Companion.seconds

private fun timeDurationSeconds(time: Double) = time.seconds
```
Consider removing this helper function and using `.seconds` directly. 😉